### DISCUSSION TITLE
Blink Script not compiling

### DISCUSSION THREAD

**Discussion by Vi21gn7es2004h** [stm32duino/Arduino_Core_STM32#2105] [Q&A] [answered]
FQBN: STMicroelectronics:stm32:GenF4:pnum=GENERIC_F407VGTX,usb=CDCgen
Using board 'GenF4' from platform in folder: C:\Users\Pragadeesh.DESKTOP-IC3LO5Q\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0
Using core 'arduino' from platform in folder: C:\Users\Pragadeesh.DESKTOP-IC3LO5Q\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0

"C:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\tools\\STM32Tools\\2.2.1/win/busybox.exe" sh "C:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/extras/prebuild.sh" "C:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Temp\\arduino\\sketches\\EB913B5F36CDEA9FF5D2A0324B8843BE" "C:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Temp\\.arduinoIDE-unsaved2023715-1244-aw6ika.tubr6\\sketch_aug15a" "C:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0"
Detecting libraries used...
"C:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\tools\\xpack-arm-none-eabi-gcc\\12.2.1-1.2/bin/arm-none-eabi-g++" -mcpu=cortex-m4 -mfpu=fpv4-sp-d16 -mfloat-abi=hard -DVECT_TAB_OFFSET=0x0 -DUSE_FULL_LL_DRIVER -mthumb "@C:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Temp\\arduino\\sketches\\EB913B5F36CDEA9FF5D2A0324B8843BE/sketch/build.opt" -c -Os -DNDEBUG -w -std=gnu++17 -ffunction-sections -fdata-sections -fno-threadsafe-statics --param max-inline-insns-single=500 -fno-rtti -fno-exceptions -fno-use-cxa-atexit "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Temp\\.arduinoIDE-unsaved2023715-1244-aw6ika.tubr6\\sketch_aug15a" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\cores\\arduino/avr" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\cores\\arduino/stm32" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\cores\\arduino/stm32/LL" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\cores\\arduino/stm32/usb" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\cores\\arduino/stm32/OpenAMP" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\cores\\arduino/stm32/usb/hid" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\cores\\arduino/stm32/usb/cdc" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Drivers/STM32F4xx_HAL_Driver/Inc" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Drivers/STM32F4xx_HAL_Driver/Src" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/STM32F4xx" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Middlewares/ST/STM32_USB_Device_Library/Core/Inc" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Middlewares/ST/STM32_USB_Device_Library/Core/Src" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Middlewares/OpenAMP" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Middlewares/OpenAMP/open-amp/lib/include" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Middlewares/OpenAMP/libmetal/lib/include" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Middlewares/OpenAMP/virtual_driver" -w -x c++ -E -CC -DSTM32F4xx -DARDUINO=10607 -DARDUINO_GENERIC_F407VGTX -DARDUINO_ARCH_STM32 "-DBOARD_NAME=\"GENERIC_F407VGTX\"" "-DVARIANT_H=\"variant_generic.h\"" -DSTM32F407xx -DUSBCON -DUSBD_VID=0 -DUSBD_PID=0 -DHAL_PCD_MODULE_ENABLED -DUSBD_USE_CDC -DHAL_UART_MODULE_ENABLED "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\tools\\CMSIS\\5.7.0/CMSIS/Core/Include/" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Drivers/CMSIS/Device/ST/STM32F4xx/Include/" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\system/Drivers/CMSIS/Device/ST/STM32F4xx/Source/Templates/gcc/" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\tools\\CMSIS\\5.7.0/CMSIS/DSP/Include" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\tools\\CMSIS\\5.7.0/CMSIS/DSP/PrivateInclude" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\cores\\arduino" "-IC:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Arduino15\\packages\\STMicroelectronics\\hardware\\stm32\\2.6.0\\variants\\STM32F4xx\\F407V(E-G)T_F417V(E-G)T" "C:\\Users\\Pragadeesh.DESKTOP-IC3LO5Q\\AppData\\Local\\Temp\\arduino\\sketches\\EB913B5F36CDEA9FF5D2A0324B8843BE\\sketch\\sketch_aug15a.ino.cpp" -o nul
Alternatives for bits/c++config.h: []
ResolveLibrary(bits/c++config.h)
  -> candidates: []
In file included from c:\users\pragadeesh.desktop-ic3lo5q\appdata\local\arduino15\packages\stmicroelectronics\tools\xpack-arm-none-eabi-gcc\12.2.1-1.2\arm-none-eabi\include\c++\12.2.1\stdlib.h:36,
                 from C:\Users\Pragadeesh.DESKTOP-IC3LO5Q\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/wiring.h:24,
                 from C:\Users\Pragadeesh.DESKTOP-IC3LO5Q\AppData\Local\Arduino15\packages\STMicroelectronics\hardware\stm32\2.6.0\cores\arduino/Arduino.h:36,
                 from C:\Users\Pragadeesh.DESKTOP-IC3LO5Q\AppData\Local\Temp\arduino\sketches\EB913B5F36CDEA9FF5D2A0324B8843BE\sketch\sketch_aug15a.ino.cpp:1:
c:\users\pragadeesh.desktop-ic3lo5q\appdata\local\arduino15\packages\stmicroelectronics\tools\xpack-arm-none-eabi-gcc\12.2.1-1.2\arm-none-eabi\include\c++\12.2.1\cstdlib:41:10: fatal error: bits/c++config.h: No such file or directory
   41 | #include <bits/c++config.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.

exit status 1

Compilation error: exit status 1

**Comment by fpistm** [comment, marked as answer]
Seems you have a too long path. It is a known issue. Unfortunately it is linked to Arduino and can't do othing. Try to install in shorten path using arduino- cli.yaml.
